File: python/leetcode/733-flood-fill.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:

        ROWS, COLS = len(image), len(image[0])
        color = image[sr][sc]

        if image[sr][sc] == newColor:
            return image

        def fill(r, c):
            if r < 0 or r >= ROWS or c < 0 or c >= COLS:
                return
            elif image[r][c] == color:
                image[r][c] = newColor
                fill(r+1, c)
                fill(r-1, c)
                fill(r, c+1)
                fill(r, c-1)

        fill(sr, sc)
        return image

File: c#/leetcode/733-flood-fill.cs
public class Solution
{
  public int[][] FloodFill(int[][] image, int sr, int sc, int newColor)
  {
    int currentColor = image[sr][sc];
    if (image[sr][sc] == newColor)
    {
      return image;
    }
    fill(image, sr, sc, currentColor, newColor);
    return image;
  }

  public void fill(int[][] image, int r, int c, int currentColor, int newColor)
  {
    int ROWS = image.Length, COLS = image[0].Length;

    if (r < 0 || r >= ROWS || c < 0 || c >= COLS || image[r][c] != currentColor)
    {
      return;
    }
    image[r][c] = newColor;
    fill(image, r, c + 1, currentColor, newColor);
    fill(image, r, c - 1, currentColor, newColor);
    fill(image, r + 1, c, currentColor, newColor);
    fill(image, r - 1, c, currentColor, newColor);
  }

}